### DISCUSSION TITLE
[companion project] HTML-like format for writing Node-Gtk UIs

### DISCUSSION THREAD

**Discussion by binyamin** [romgrk/node-gtk#267] [Ideas]
> Would anyone use this?

**Problem**: I have a lot more trouble writing Gtk UIs than writing webpages. It's tedious to write a GtkBuilder UI definition in XML, and many developers seem to use a GUI app like [Glade](https://glade.gnome.org/).

**Proposal**: Create a file-format similar to HTML, using Gtk widget names and properties. Then, write a library which accepts a file path and returns a GtkBuilder Object.

**Example**
```html
<window>
    <box orientation="vertical">
        <button onClick="() => doSomething()">Click me</button>
        <label id="label1">Hello World</label>
    </box>
</window>
```

**Comment by romgrk** [comment]
I must say glade's format is quite annoying, way too verbose to be comfortable to hand-write. However, creating an alternative HTML-like format seems a bit involved. I'm curious, why would you go with a HTML format rather than using/expanding on react-gtk?

**Reply by binyamin** [reply]
Honestly, I just don't like using react. It's just another layer of code. Either way, spitting out a GtkBuildable seems much more efficient.

**Reply by romgrk** [reply]
Sounds fair, I enjoy React a lot but I can understand the reticence to adding layers of code. If the project is well done I'd be happy to link or include it.
